locator/src/control_plane.rs

```diff
@@ -136,7 +136,7 @@ mod tests {
     #[tokio::test]
     async fn test_control_plane() {
         let _server = TestControlPlaneServer::spawn("127.0.0.1", 9000).unwrap();
-        std::thread::sleep(Duration::from_millis(100));
+        std::thread::sleep(Duration::from_millis(300));
```

Comment:
I wonder if there's a better way to find out whether the server is up or not. Maybe some sort of /ready in the test setup which is checked before moving on with the tests?

Reply:
Yeah.. i just noticed this started failing a lot. I added a fix for this here https://github.com/getsentry/synapse/pull/43 which waits on the tcp connection inside spawn() method